CSS/css_completions.py

```diff
@@ -714,14 +925,19 @@ def complete_property_value(self, view, prefix, pt):
 
         return completions
 
-    def complete_function_argument(self, view, prefix, pt):
-        args_region = view.expand_by_class(
-            pt, sublime.CLASS_PUNCTUATION_START | sublime.CLASS_PUNCTUATION_END)
-        func_region = view.expand_by_class(
-            args_region.a - 2, sublime.CLASS_WORD_START | sublime.CLASS_WORD_END)
-        func_name = view.substr(func_region)
-
-        # TODO: provide more function specific argument completions
+    def complete_function_argument(self, view: sublime.View, prefix, pt):
+        func_name = ""
+        nest_level = 1
+        for i in range(pt - 1, pt - 32 * 1024, -1):
+            ch = view.substr(i)
+            if ch == ")" and not view.match_selector(i, "string, comment"):
```

Comment:
I'd vote for moving the `view.match_selector(i, "string, comment")` call to a line above to avoid it needing to be repeated, and `continue`ing if it's `True`.

Can we have some comments or more explanation of how this method works, please? I understand it looks backwards from the given point one char at a time to find where the function name is, but I don't entirely understand how/the nesting level stuff as it's slightly backwards from what I'm used to (probably because it looks backwards?). It's s shame we don't have unit tests because that would likely clarify it enough for me.

Reply:
The current implementation calls `view.match_selector()` only if `ch` is a parenthesis. Splitting that statement and moving `view.match_selector()` would cause it to be called on each charachter, which would significantly slow down the function.

If someone can come up with a more pythonic solution, I'd be happy to use that instead.

The main issue is a function can be called in another function's arguments list. The level of nesting is undefined. So we need to make sure to ignore any function call arguments list located in front of the caret.

Example

```
.text-calc-clamp-max-min {
    top: clamp(1*5pt, calc(12/5), | 100rem);
}
```

The `|` denotes caret. We want to make sure to return `clamp` rather than `calc`

and do so even if parentheses occure in strings. Not sure if it can happen in CSS, but wanted to make sure it works with it.

```
.class {
    top: any(foo(bar(baz("(")),  ) |  );
}
```

The function basically finds the next opeing parentheses while ignoring any balanced parens.

Reply:
> The current implementation calls `view.match_selector()` only if `ch` is a parenthesis. Splitting that statement and moving `view.match_selector()` would cause it to be called on each charachter, which would significantly slow down the function.

I forgot about short circuiting here, thanks for correcting me :)

Nice, thanks for the explanation and the comments